src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -322,6 +322,11 @@ public new virtual SqliteDataReader ExecuteReader(CommandBehavior behavior)
                     ? PrepareAndEnumerateStatements()
                     : _preparedStatements)
                 {
+                    if (!unprepared)
+                    {
+                        raw.sqlite3_reset(stmt);
```

Comment:
Okay I moved the reset behind the stepping (statements passed to the reader are already being reset).